torch/_dynamo/eval_frame.py

```diff
@@ -971,7 +968,12 @@ def dynamo_normalization_capturing_compiler(
             graph = gm
 
             nonlocal fake_mode, example_inputs
-            fake_mode = fake_mode or _guards.detect_fake_mode(inner_example_inputs)
+            # NB: do NOT pass inner_example_inputs here, we are detecting the
+            # Dynamo allocated fake mode, which should be DISTINCT from a
+            # potential outer ambient fake mode which the user provided.
```

Comment:
>          # Dynamo allocated fake mode, which should be DISTINCT from a
>          # potential outer ambient fake mode which the user provided.

IIUC, we are changing the original approach for exporting fake model? 

The original idea was to allow users to create their own Fake Mode which would be reused by Dynamo internally. That forces the user to instantiate a fake mode with ShapeEnv if they care about dynamic shapes.

Now it seems the user can fakefy input/model with their own fake mode, but `torch._dynamo.export` will still create another fake mode. My question is: How creating a second fake mode would handle input/parameters already fakefied by another instance? I thought we couldn't mix a tensor fakefied with one fake mode with a different fake mode instance

Reply:
This is what the updated PR does. `allow_non_fake_inputs` is generalized to mean "allow non-fake inputs, and unrelated fake inputs that aren't for your mode"